lib/can/src/CANInterface.cpp

```diff
@@ -0,0 +1,157 @@
+#include "CANInterface.h"
+
+CANInterface::CANInterface(const Config &config)
+    : m_CANBus1(config.can1_RX, config.can1_TX, config.frequency_hz),
+      m_CANBus2(config.can2_RX, config.can2_TX, config.frequency_hz),
+      m_activeCANBus(&m_CANBus1),
+      m_rxPostmanThread(RX_POSTMAN_THREAD_PRIORITY),
+      m_rxClientThread(RX_CLIENT_THREAD_PRIORITY),
+      m_txProcessorThread(TX_PROCESSOR_THREAD_PRIORITY),
+      m_rxStreamedMsgMap(config.rxStreamedMsgMap),
+      m_txStreamedMsgMap(config.txStreamedMsgMap),
+      m_rxOneShotMsgHandler(config.rxOneShotMsgHandler) {
+  // Set hw filters
+  m_CANBus1.setFilter(config.filterID, config.filterFormat, config.filterMask);
+  m_CANBus2.setFilter(config.filterID, config.filterFormat, config.filterMask);
+
+  // Put CAN bus 2 in silent monitoring mode
+  m_CANBus2.monitor(true);
```

Comment:
Need to test what this does, Mbed docs say it puts the CAN bus into "silent monitoring mode"